src/js/components/mixins/Parse.jsx

```diff
@@ -1,29 +1,8 @@
 'use strict';
-var model = require('../../model'),
-    sg = require('../../model/signals'),
-    hierarchy = require('../../util/hierarchy');
+var model = require('../../model');
 
 module.exports = {
   parse: function(primitive) {
-    model.parse().then(function() {
-      // @TODO: long-term this logic ought to be handled through the redux store,
-      // but for now we need to re-broadcast a Lyra signal to keep the selected
-      // mark selected.
-
-      // Walk up from the selected primitive to create an array of its parent groups' IDs
-      var parentGroupIds = hierarchy.getParentGroupIds(model.lookup(primitive._id)),
-          sceneRoot = model.view.model().scene().items[0],
-          item;
-
-      // Create a path array of group layer IDs (inclusive of the selected layer),
-      // then walk down the rendered Lyra scene to find a corresponding item.
-      item = hierarchy.findInItemTree(sceneRoot, [primitive._id].concat(parentGroupIds));
-      // If an item was found, set the Lyra mode signal so that the handles appear.
-      // As noted above, this logic should probably not exist here!
-      if (item !== null) {
-        sg.value(sg.SELECTED, item);
-        model.update();
-      }
-    });
+    model.parse();
```

Comment:
We do need this for now, because a re-parse of the model is needed in order to have certain updates take effect. I absolutely support removing the parse mixin in favor of an action that triggers that re-parse down the road, it simply felt like it was out of scope for this PR.